src/Toolkit/Toolkit/UI/Controls/FeatureForm/ComboboxFormInputView.cs

```diff
@@ -0,0 +1,140 @@
+
+        /// <summary>
+        /// Initializes an instance of the <see cref="ComboBoxFormInputView"/> class.
+        /// </summary>
+        public ComboBoxFormInputView()
+        {
+            DefaultStyleKey = typeof(ComboBoxFormInputView);
+        }
+
+        /// <inheritdoc />
+        public override void OnApplyTemplate()
+        {
+            base.OnApplyTemplate();
+            if (_selector != null)
+            {
+                _selector.SelectionChanged -= Selector_SelectionChanged;
+            }
+            _selector = GetTemplateChild("Selector") as System.Windows.Controls.Primitives.Selector;
+            if(_selector != null)
+            {
+                _selector.SelectionChanged += Selector_SelectionChanged;
+            }
+            UpdateItems();
+        }
+
+        /// <summary>
+        /// Gets or sets the FieldFormElement.
+        /// </summary>
+        public FieldFormElement? Element
+        {
+            get { return (FieldFormElement)GetValue(ElementProperty); }
+            set { SetValue(ElementProperty, value); }
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="Element"/> dependency property.
+        /// </summary>
+        public static readonly DependencyProperty ElementProperty =
+            DependencyProperty.Register(nameof(Element), typeof(FieldFormElement), typeof(ComboBoxFormInputView), new PropertyMetadata(null, (s, e) => ((ComboBoxFormInputView)s).OnElementPropertyChanged(e.OldValue as FieldFormElement, e.NewValue as FieldFormElement)));
+
+        private void OnElementPropertyChanged(FieldFormElement? oldValue, FieldFormElement? newValue)
+        {
+            if (oldValue is INotifyPropertyChanged inpcOld)
+            {
+                _elementPropertyChangedListener?.Detach();
+                _elementPropertyChangedListener = null;
+            }
+            if (newValue is INotifyPropertyChanged inpcNew)
+            {
+                _elementPropertyChangedListener = new WeakEventListener<ComboBoxFormInputView, INotifyPropertyChanged, object?, PropertyChangedEventArgs>(this, inpcNew)
+                {
+                    OnEventAction = static (instance, source, eventArgs) => instance.Element_PropertyChanged(source, eventArgs),
+                    OnDetachAction = static (instance, source, weakEventListener) => source.PropertyChanged -= weakEventListener.OnEvent,
+                };
+                inpcNew.PropertyChanged += _elementPropertyChangedListener.OnEvent;
+            }
+            UpdateItems();
+        }
+
+        private void Element_PropertyChanged(object? sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == nameof(FieldFormElement.Value))
+            {
+                if (Dispatcher.CheckAccess())
+                    UpdateSelection();
+                else
+                    Dispatcher.Invoke(UpdateSelection);
```

Comment:
Doesn't Dispatcher.Invoke already do this check?  I'm pretty sure it executes immediately if we are on the right thread.  Or did you mean to do immediate-or-BeginInvoke behavior, to avoid blocking the INPC-raising thread?

Same question for all the other implementations of Element_PropertyChanged in this PR.

Reply:
When performing async evaluation of expressions, core might be coming back on a different thread, so sometimes we need to dispatch back to the UI thread, and but many other times we do not. You're correct [there's some logic](https://github.com/dotnet/wpf/blob/873db676363ce1777e763948481b24ded3513746/src/Microsoft.DotNet.Wpf/src/WindowsBase/System/Windows/Threading/Dispatcher.cs#L579) to avoid extra work, but this is the pattern we have followed everywhere.